### PR TITLE
Edit metadata translations in the UI: per-entity side panel + translate-or-rename prompt

### PR DESCRIPTION
**Pull request by FelixMalfait**
## Context

The resolution model already supported per-locale workspace translations (`overrides.translations[locale][property]`), but nothing wrote that key and nothing exposed it — every label edit was implicitly a canonical rename. This ships the editing experience discussed in the mockups: **two write operations (rename vs translate), your locale decides what you see but never what you may edit.**

## Backend

- **`metadataTranslations` query** (new `metadata-translation` module, `DATA_MODEL`-guarded): per-(property × locale) rows with `sourceValue`, `canonicalValue`, resolved `value`, and `provenance` (`WORKSPACE` / `SHIPPED` / `INHERITED`). Scoped to one entity: all locales for the side panel, or a single locale when a caller only needs the viewer's resolution. `value` is computed by `resolveEffectiveEntityProperty` itself, so the editor can never disagree with what the app renders.
- **Write path**: `updateOneObject` / `updateOneField` accept `translations: [{ locale, property, value }]`; `mergeTranslationsIntoOverrides` mirrors `computeMetadataOverridesBlob` semantics (empty value deletes, empty groups prune, empty blob collapses to null). Properties validated against the shared translatable registry.
- **Custom entities carry translations too**: the resolver checks workspace translations before the custom-entity short-circuit, and the sanitizers preserve the overrides blob instead of nulling it on custom updates — a custom object authored in French can now be translated into English. One rule for standard, custom, and app entities. Parity spec frozen resolvers amended accordingly.

## Frontend

- **Side panel** (the settings side panel we wanted to introduce — one new `SidePanelPages` entry on the existing side-panel infrastructure): opened from "Edit translations" on the object Settings tab and field edit page. Each property gets a standard `H2Title` section whose description shows the source text once ("Source: Company"); the rows below are translations only — language, click-to-edit value, and a reset button on workspace-edited rows as the sole state marker. Untranslated languages show the fallback as ghost text, with one explanatory line at the top. For an entity authored in a non-source language, the source slot shows the authored text and every locale row (the source language included) is a plain translation row — no special rows, no provenance vocabulary.
- **Translate-or-rename prompt**: when a label edit goes through a translation (resolved ≠ canonical for the viewer's locale), a `ConfirmationModal` in the standard two-stacked-buttons layout (same shape as the billing modals) asks "Only in {language}" (primary, default) vs "Rename for all languages" (secondary). No Cancel button — click-outside/Escape dismisses, like elsewhere. No prompt when editing the canonical value — the common gesture stays one click. With label↔API-name sync enabled, the translate path drops the sync-derived name fields so a translation never renames the API.
- **Bug fix that fell out**: the object About form sent *all* form values on any blur-save, so a non-English viewer editing e.g. the icon would silently push their resolved (translated) labels as canonical renames. It now sends dirty fields only, and the save state machine lives in `useSaveUpdateDataModelObjectAboutForm`.
- `useLocaleOptions` extracted from `LocalePicker` (language names were previously inline there).

Two things were built here and then taken back out in review, both deliberately:

- A workspace-wide Settings › Translations page — the per-entity side panel is the editing surface for now, and the backend query is entity-scoped only.
- A shared-layout fix giving collapsed `AnimatedExpandableContainer`s no flex-gap footprint (`display: contents` on the root). It changes the layout contract for ~24 consumers, which is a separate logical change from metadata translations; it now lives alone on `claude/uncompiled-message-catalog-m3daw3-s21`. The phantom gap it fixes predates this PR — the two `AdvancedSettingsWrapper` sections on the object Settings tab already claimed a gap slot each while collapsed — so nothing here regresses by leaving it out.

## Validation

- End-to-end against the seeded env: write → `WORKSPACE` provenance wins, canonical untouched; reset → falls back to `SHIPPED`; catalog precedence intact. Sync-on custom object verified live: translate-only save leaves API names and canonical label untouched.
- Server: touched suites 47/47 (incl. override parity corpus, the S5 guard spec, and specs for both new pure utils), `tsgo` clean, oxlint clean.
- Front: `tsgo` clean, oxlint + oxfmt clean, affected suites green; Storybook stories per new component.
- Screenshots of both surfaces (French session included) shared in the working thread.

> `api-breaking-changes` is red for a reason unrelated to this PR: `@graphql-inspector/cli@7.0.0` was published upstream today depending on `@graphql-inspector/coverage-command@^7.0.0`, which does not exist, so the workflow's unpinned global install fails `ETARGET` on every PR in the repo. Details and a one-line pin are in [this comment](https://github.com/twentyhq/twenty/pull/24374#issuecomment-5355676559).

## Follow-ups (deliberately out of scope)

- Stale markers (translated-before-rename) need authored-at metadata on translations.
- AI-fill of missing translations (slots in as entity translations with their own provenance).
- Per-locale translation permission split from `DATA_MODEL` if roles should allow language-only editors.
- Enabling label↔API-name sync while viewing through a translation derives names from the displayed labels and is rejected server-side; the clean fix is server-computed names when sync is on.
- Reintroduce a workspace-wide translations overview page once there's a clear job for it.

### CODE REVIEW

File: packages/twenty-front/src/modules/settings/data-model/object-details/components/SettingsUpdateDataModelObjectAboutForm.tsx [hunk 1]
```diff
@@ -140,10 +214,58 @@ export const SettingsUpdateDataModelObjectAboutForm = ({
...
+    const translations = dirtyTranslatableProperties.map((property) => ({
+      locale: currentLocale,
+      property,
+      value: pendingFormValues[property] ?? null,
+    }));
+
+    const updateResult = await updateOneObjectMetadataItem({
+      idToUpdate: objectMetadataItem.id,
+      updatePayload: { translations },
+    });
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/data-model/object-details/components/SettingsUpdateDataModelObjectAboutForm.tsx]: Fixed in fac45f85 — the translate-only branch now sends unrelated dirty values (icon, …) as canonical updates in the same mutation, so the form reset reflects what was actually saved. `pickDirtyValues` is shared between both save paths.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-server/src/engine/metadata-modules/metadata-translation/services/metadata-translation.service.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+      return isDefined(flatFieldMetadata)
+        ? [toFieldEntity(flatFieldMetadata)]
+        : [];
+    }
+
+    const activeObjects = Object.values(
+      flatObjectMetadataMaps.byUniversalIdentifier,
+    )
+      .filter(isDefined)
+      .filter(({ isActive }) => isActive);
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-server/src/engine/metadata-modules/metadata-translation/services/metadata-translation.service.ts]: Fixed in fac45f85, suggestion applied — workspace-wide selection now excludes system objects (fields already excluded system + inactive), and the front's redundant client-side filter was removed since the backend is now the single rule.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-server/src/engine/metadata-modules/utils/merge-translations-into-overrides.util.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  );
+
+  for (const { locale, property, value } of translationEntries) {
+    if (isNonEmptyString(value)) {
+      translations[locale] = { ...translations[locale], [property]: value };
+      continue;
+    }
+
+    if (isDefined(translations[locale])) {
+      delete translations[locale][property];
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-server/src/engine/metadata-modules/utils/merge-translations-into-overrides.util.ts]: Hardened in fac45f85: locale and property are already allowlist-validated upstream (`@IsIn(APP_LOCALES)` and the translatable-properties registry check in the sanitizers), and the merge util now additionally filters `__proto__`/`constructor`/`prototype` keys as defense in depth so no assignment path can reach the prototype chain.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-server/src/engine/metadata-modules/metadata-translation/services/metadata-translation.service.ts [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  }: {
+    input: MetadataTranslationsInput;
+    workspaceId: string;
+  }): Promise<MetadataTranslationDTO[]> {
+    const entities = await this.collectTranslatableEntities({
+      input,
+      workspaceId,
+    });
+    const locales = isDefined(input.locale)
+      ? [input.locale]
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-server/src/engine/metadata-modules/metadata-translation/services/metadata-translation.service.ts:94]: Partially applied in fac45f85: the fully unbounded all-entities × all-locales shape is now rejected (`UserInputError` when neither entity id nor locale is given). The one-entity × all-locales fan-out is the side panel's product shape — every language on one screen — and each locale's catalog fetch is a cache read batched per application, so I kept it rather than paginating the panel.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-front/src/pages/settings/translations/SettingsTranslations.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    useFilteredObjectMetadataItems();
+  const { metadataTranslations, refetch } = useMetadataTranslations({
+    locale: selectedLocale,
+  });
+  const { saveMetadataTranslation } = useSaveMetadataTranslation();
+  const { getPropertyLabel } = useTranslatablePropertyLabel();
+
+  const rowsByOwnerId = new Map<string, MetadataTranslationRow[]>();
+
+  for (const row of metadataTranslations) {
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/pages/settings/translations/SettingsTranslations.tsx]: Save flow moved into `useMetadataTranslations` (shared with the side panel) in fac45f85. The remaining grouping/search code is render-shaping for this page's table layout; kept in place to avoid a single-consumer indirection, following how SettingsObjectFieldTable structures its own rows in `src/pages`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by FelixMalfait** [comment]
All 9 inline findings addressed in fac45f85 (system-object exclusion, translate-only keeping unrelated edits, prototype-pollution hardening, unbounded-query guard, typed service, shared save flow, comment/cast nits). On the two high-level points, which are product calls:

- **Bundling** — the three surfaces were built together deliberately (per the design discussion) because they share exactly two write operations and one query; splitting them now would ship a query with no consumer or a panel with no write path. Happy to split if preferred.
- **Feature flag** — the surfaces are DATA_MODEL-gated but ship enabled. Gating the nav item, the two "Edit translations" buttons, and the intent prompt behind a `FeatureFlagKey` is a ~30-line change if a gradual rollout is wanted — say the word.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by FelixMalfait** [comment]
CI status on the last red round, all three causes identified:

- `front-task (lint)` / `server-lint-typecheck`: unformatted files from my review-fix commit — fixed and pushed (b7ec1f90), which also regenerates the `twenty-client-sdk` metadata schema the `server-validation` drift check flagged.
- `create-app-e2e-minimal`: **not this PR** — upstream `rolldown@1.2.5` was published at 13:33 UTC with its `@rolldown/binding-darwin-x64` binding missing from npm, and the scaffolded test app resolves fresh (the template ships an empty lockfile), so every resolution since then fails with `No candidates found`. A sibling run at 13:10 was green with identical steps; most other PRs skip this job entirely (it's gated on SDK-adjacent paths, which this PR's `twenty-shared` enum additions trip). It heals with a re-run once upstream ships the missing binding or 1.2.6. Committing a real `yarn.lock` in `packages/create-twenty-app/src/constants/template/` would immunize CI against this class of upstream breakage — happy to do that as a separate PR.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-server/src/engine/metadata-modules/metadata-translation/dtos/metadata-translations.input.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { Field, InputType } from '@nestjs/graphql';
+
+import { APP_LOCALES } from 'twenty-shared/translations';
+import { IsIn, IsOptional, IsUUID } from 'class-validator';
+
+import { UUIDScalarType } from 'src/engine/api/graphql/workspace-schema-builder/graphql-types/scalars';
+
+// Two access shapes, one query: an entity id scopes to that entity across all
+// locales (the per-entity translations panel), a locale alone scopes to every
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-server/src/engine/metadata-modules/metadata-translation/dtos/metadata-translations.input.ts]: Fixed in b753f6a0 — the comment now describes the entity-scoped shapes only. The remaining `if (!isDefined(input.fieldMetadataId)) return []` is the type-level narrowing for the guard that lives in `findMetadataTranslations`, not a reachable access shape.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-server/src/engine/metadata-modules/metadata-translation/services/metadata-translation.service.ts [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+          translations.push({
+            metadataName,
+            recordId,
+            objectMetadataId,
+            property,
+            locale,
+            sourceValue,
+            canonicalValue,
+            value,
+            provenance: isDefined(workspaceTranslation)
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-server/src/engine/metadata-modules/metadata-translation/services/metadata-translation.service.ts]: Fixed in 910a517b — extracted `resolveProvenance` as an if-chain helper.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-front/src/modules/settings/translations/components/SidePanelSettingsMetadataTranslationsPage.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+  font-size: ${themeCssVariables.font.size.sm};
+`;
+
+export const SidePanelSettingsMetadataTranslationsPage = () => {
+  const { t } = useLingui();
+  const settingsTranslationsSidePanelTarget = useAtomStateValue(
+    settingsTranslationsSidePanelTargetState,
+  );
+  const { metadataTranslations, saveTranslationRow } = useMetadataTranslations(
+    isDefined(settingsTranslationsSidePanelTarget)
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/translations/components/SidePanelSettingsMetadataTranslationsPage.tsx:76]: Fixed in 910a517b — the target→input mapping now lives in a `getMetadataTranslationsInput` helper; the hook call keeps a single flat ternary for the null (panel closed) case since the hook must be called unconditionally.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-front/src/modules/settings/translations/components/MetadataTranslationValueCell.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
+import { type MetadataTranslationRow } from '@/settings/translations/hooks/useMetadataTranslations';
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/translations/components/MetadataTranslationValueCell.tsx:1]: Added in 783e0441 — one story per component, each with a play/findBy assertion (the cell also covers the click-to-edit → save path, the page mocks the metadataTranslations query and asserts the Source/Edited tags). The one-util-per-file nit from the summary is also addressed in fce791d1 (`findInvalidTranslationOverrideProperties` split into its own util with a shared type file).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-front/src/modules/settings/data-model/object-details/components/SettingsUpdateDataModelObjectAboutForm.tsx [hunk 10]
```diff
@@ -9,24 +11,44 @@ import {
...
 
+const TRANSLATION_INTENT_MODAL_ID = 'object-label-translation-intent-modal';
+
+const OBJECT_TRANSLATABLE_PROPERTIES =
+  TRANSLATABLE_PROPERTIES_BY_METADATA_NAME.objectMetadata;
+
+type ObjectTranslatableProperty =
+  (typeof OBJECT_TRANSLATABLE_PROPERTIES)[number];
+
 export const SettingsUpdateDataModelObjectAboutForm = ({
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/data-model/object-details/components/SettingsUpdateDataModelObjectAboutForm.tsx:28]: Done in 2ebd1a97 — the save state machine (both save paths, intent prompt trigger, pending state, dirty-field picking) now lives in `useSaveUpdateDataModelObjectAboutForm`; the component is 107 lines and only builds the form and renders.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/data-model/object-details/components/SettingsUpdateDataModelObjectAboutForm.tsx:28]: Intentional, and requested in this PR's review: the oversized gap before Danger zone on the object Settings tab (reported by the PR author) is caused by collapsed advanced sections leaving an empty root div that claims flex-gap slots. That failure mode is not specific to this page — every consumer that puts an AdvancedSettingsWrapper inside a gap container has phantom spacing while collapsed — so patching it per-consumer would re-create the bug everywhere else.

What the change does: nothing when a section is expanded (the panel keeps its own box, sizes, and animation); collapsed sections stop reserving gap slots, which is the fix. Any Argos diffs on other stories should show exactly that tightening and nothing else — flagging them for human approval is the right mechanism here. Verified by hand on the object settings tab (both custom and standard objects, advanced on and off) in the working thread's screenshots.

If the Argos diffs surface anything beyond collapsed-state spacing, I'll pull the change out into its own PR instead.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-ui/src/layout/AnimatedExpandableContainer/AnimatedExpandableContainer.module.scss [hunk 11]
```diff
@@ -1,3 +1,10 @@
+// The root must not take layout space of its own: when the panel is closed
+// (and unmounted), a flex parent with gap would still leave a gap slot for
+// the empty root div.
+.root {
+  display: contents;
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-ui/src/layout/AnimatedExpandableContainer/AnimatedExpandableContainer.module.scss]: Intentional, and requested in this PR's review: the oversized gap before Danger zone on the object Settings tab (reported by the PR author) is caused by collapsed advanced sections leaving an empty root div that claims flex-gap slots. That failure mode is not specific to this page — every consumer that puts an AdvancedSettingsWrapper inside a gap container has phantom spacing while collapsed — so patching it per-consumer would re-create the bug everywhere else.

What the change does: nothing when a section is expanded (the panel keeps its own box, sizes, and animation); collapsed sections stop reserving gap slots, which is the fix. Any Argos diffs on other stories should show exactly that tightening and nothing else — flagging them for human approval is the right mechanism here. Verified by hand on the object settings tab (custom and standard objects, advanced on and off) in the working thread's screenshots.

If the Argos diffs surface anything beyond collapsed-state spacing, I'll pull this shared-layout fix out into its own PR instead. (Apologies for the copy of this reply that landed on the resolved about-form thread — wrong anchor.)

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-ui/src/layout/AnimatedExpandableContainer/AnimatedExpandableContainer.module.scss]: The visual regression review is in and confirms the intended scope: 10 changes, all explained by this PR — the 5 changed stories are precisely the AnimatedExpandableContainer consumers (nav drawer, REST playground, graph widget, AI chat thinking steps) with small collapsed-state spacing diffs (1–12%), and the 5 additions are the new translation-component stories. Nothing beyond the intended tightening, so the fix stays in this PR. Resolving.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-ui/src/layout/AnimatedExpandableContainer/AnimatedExpandableContainer.module.scss [hunk 12]
```diff
@@ -1,3 +1,10 @@
+// The root must not take layout space of its own: when the panel is closed
+// (and unmounted), a flex parent with gap would still leave a gap slot for
+// the empty root div.
+.root {
```
Comment (FelixMalfait) [inline packages/twenty-ui/src/layout/AnimatedExpandableContainer/AnimatedExpandableContainer.module.scss]: Not a big fan of this kind of global change. Do we really need it? What are our other options that stay within current patterns? (even if it implies a bigger refactor)

Comment (FelixMalfait) [reply packages/twenty-ui/src/layout/AnimatedExpandableContainer/AnimatedExpandableContainer.module.scss]: Pulled it out of this PR (83e2d951). It now sits alone on `claude/uncompiled-message-catalog-m3daw3-s21`, ready to open as its own PR whenever you want it — say the word. The bot raised the same "one PR = one logical change" objection earlier and I argued it down on the strength of clean Argos diffs; you asking independently means I got that call wrong, and the diffs were never the real issue.

Two things I found while digging that change the framing:

**This PR doesn't need the fix.** The gap you screenshotted isn't caused by the Translations section — it predates this PR. `ObjectSettings`'s container is `flex column; gap: spacing[8]`, and two of its children are `<AdvancedSettingsWrapper>` (Search, Indexes). Advanced mode is off by default, so each of those leaves an empty root div that still claims a gap slot: three gaps where you want one. On main today that same triple gap sits after "Options" — I just inserted Translations in front of it, which is why it started looking wrong. So dropping the fix here costs nothing; this PR is now exactly as good as main on that front.

**It's not really about `AdvancedSettingsWrapper` either.** Any always-mounted wrapper inside a gapped flex parent does this, and `AnimatedExpandableContainer` has ~24 consumers. That's why I reached for the shared fix — but it's also why it deserves its own review.

The three real options, since you asked:

1. **`display: contents` on the root** — what's on s21. One rule, fixes every occurrence. The argument for it: that root is a *grouping* element, not a layout element, and this is the CSS that says so. I checked that nothing depends on it being a box — zero `styled(AnimatedExpandableContainer)`, no consumer targets the wrapper by selector. Against it: it silently changes layout semantics for 24 consumers at once, which is your objection and a fair one.

2. **Don't mount the wrapper while collapsed** — `AnimatedExpandableContainer` returns `null` once `!isExpanded` and the exit transition has ended (tracked via `onTransitionEnd`, so no `useEffect`). No global CSS semantics change: whenever anything is visible the box behaves exactly as today, and when nothing is visible there's no element to claim a slot. Costs ~15 lines of transition bookkeeping in a shared component. This is the "bigger refactor that stays within patterns" answer, and if you want one of the two, I'd pick this one over 1.

3. **Fix per call site** — drop `gap` for `> * + * { margin-top }` where a conditionally-empty child sits. Zero blast radius, but it's 10 sites across 7 files today and the trap resets for every settings page written afterwards. I don't think this is worth it.

I did *not* find a way to avoid the wrapper div itself: base-ui's `Collapsible.Root` renders a `<div>`, the `render` prop can't merge it with `Panel`, and hand-rolling the panel loses base-ui's height measurement and transition-status attributes — more code, not less.

My recommendation is option 2 on s21, replacing what's cherry-picked there now. Want me to write it that way?

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-server/src/engine/metadata-modules/utils/find-invalid-translation-override-properties.util.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import { ALL_TRANSLATABLE_PROPERTIES_BY_METADATA_NAME } from 'src/engine/metadata-modules/flat-entity/constant/all-translatable-properties-by-metadata-name.constant';
+import { type TranslationOverrideEntry } from 'src/engine/metadata-modules/utils/translation-override-entry.type';
+
+export const findInvalidTranslationOverrideProperties = (
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-server/src/engine/metadata-modules/utils/find-invalid-translation-override-properties.util.ts:4]: Added in 60ff974a — `find-invalid-translation-override-properties.util.spec.ts`, 6 cases covering the allowlist filter, the empty-input case, per-metadata-name scoping (`labelSingular` is valid for `objectMetadata` and invalid for `fieldMetadata`), the `Set` dedup across locales, and the unknown-metadata-name `?? []` fallback rejecting every property. 6/6 pass, and the whole `utils/__tests__` folder is 47/47.

`'agent'` needed no cast for the fallback case — the registry is typed `Partial<Record<AllMetadataName, …>>`, so any metadata name that translates nothing is already assignable.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by FelixMalfait** [comment]
`api-breaking-changes` is red here but not because of this PR — it's broken repo-wide right now, and every open PR will hit it.

The job never got as far as comparing schemas. It died installing its tooling:

```
=== INSTALLING GRAPHQL INSPECTOR CLI ===
npm error code ETARGET
npm error notarget No matching version found for @graphql-inspector/coverage-command@^7.0.0
```

`ci-breaking-changes.yaml:465` runs `npm install -g @graphql-inspector/cli` unpinned. `@graphql-inspector/cli@7.0.0` was published upstream earlier today and depends on `@graphql-inspector/coverage-command@^7.0.0` — a version that was never published (the registry tops out at `6.1.14`). So the install now fails for everyone.

Timing confirms it's an upstream publish, not a code change: on #24410 this same step succeeded at **10:59:43** today; here it failed at **12:08:02**. Nothing in between touched this workflow.

For what it's worth, the two introspections were captured fine before the tooling died — core schema byte-identical to main, metadata schema larger by exactly the `metadataTranslations` addition, which is the expected additive change.

One-line fix pinning the CLI to the last installable release (`6.0.8`, whose deps are exact-pinned and which I verified installs clean) is sitting on `claude/uncompiled-message-catalog-m3daw3-s22`. I've deliberately not put it in this PR — it's unrelated CI infra and belongs on its own. Happy to open it as a PR whenever you want; it'll unblock the check on every open PR at once.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-front/src/modules/settings/translations/hooks/useMetadataTranslations.ts [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+export type MetadataTranslationRow =
+  MetadataTranslationsQuery['metadataTranslations'][number];
+
+export const useMetadataTranslations = (
+  input: MetadataTranslationsInput | null,
+) => {
+  const { data, loading, refetch } = useQuery(MetadataTranslationsDocument, {
+    variables: { input: input ?? {} },
+    skip: input === null,
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/translations/hooks/useMetadataTranslations.ts]: Fixed in 8d810318 — `skip: !isDefined(input)`, matching the guard already used lower in the same file.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/twenty-front/src/modules/settings/translations/components/MetadataTranslationValueCell.tsx [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+      fullWidth
+      autoFocus
+      value={draftValue}
+      onChange={setDraftValue}
+      onBlur={async () => {
+        setIsEditing(false);
+
+        const value = draftValue.trim() === '' ? null : draftValue;
+
+        if (value === row.value || (value === null && isInherited)) {
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/translations/components/MetadataTranslationValueCell.tsx]: Fixed in 037f887d — `!isDefined(value)`, with the `twenty-shared/utils` import added.

I also swept every non-generated file in the diff for the same pattern rather than wait for a third round: this was the last one. The only other `!== undefined` in a touched file is `sanitize-raw-update-field-input.ts:78`, which is pre-existing and outside this diff, so I left it alone.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by FelixMalfait** [comment]
Follow-up: this fixed itself upstream — **no pin needed, disregard my earlier suggestion.**

`@graphql-inspector/coverage-command@7.0.0` has now been published, so `cli@7.0.0`'s dependency resolves and the unpinned global install works again. Verified just now against the registry: `npm install -g @graphql-inspector/cli` installs clean (201 packages) and the binary runs.

None of the still-open PRs is red on this check — #24410's run predated the outage and #24409's ran after it, both green. It only ever caught this PR, in the window between roughly 11:00 and 13:30 UTC yesterday.

The branch with the one-line pin (`claude/uncompiled-message-catalog-m3daw3-s22`) is therefore obsolete. I've left it unmerged and unopened rather than deleting it, in case you'd still want the pin on principle — an unpinned global install in CI will break again the next time upstream publishes a broken release. That's a standing judgement call, not something this outage forces.

---
_Generated by [Claude Code](https://claude.ai/code)_
